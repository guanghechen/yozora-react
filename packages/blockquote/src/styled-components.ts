import 'styled-components'
import type { YozoraBlockquoteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    blockquote?: YozoraBlockquoteTheme
  }
}
