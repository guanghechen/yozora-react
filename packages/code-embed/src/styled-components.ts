import type { YozoraCodeEmbedTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeEmbed?: YozoraCodeEmbedTheme
  }
}
