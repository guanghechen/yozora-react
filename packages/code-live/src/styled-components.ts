import 'styled-components'
import type { YozoraCodeLiveTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeLive?: YozoraCodeLiveTheme
  }
}
