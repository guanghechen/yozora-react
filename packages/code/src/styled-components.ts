import 'styled-components'
import type { YozoraCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    code?: YozoraCodeTheme
  }
}
