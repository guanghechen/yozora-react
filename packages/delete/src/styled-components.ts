import type { YozoraDeleteTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    delete?: YozoraDeleteTheme
  }
}
