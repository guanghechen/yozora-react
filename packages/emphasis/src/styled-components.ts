import 'styled-components'
import type { YozoraEmphasisTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    emphasis?: YozoraEmphasisTheme
  }
}
