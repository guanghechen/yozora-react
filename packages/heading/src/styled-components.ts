import 'styled-components'
import type { YozoraHeadingTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    heading?: YozoraHeadingTheme
  }
}
