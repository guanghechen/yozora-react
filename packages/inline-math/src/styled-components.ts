import 'styled-components'
import type { YozoraInlineMathTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    inlineMath: YozoraInlineMathTheme
  }
}
