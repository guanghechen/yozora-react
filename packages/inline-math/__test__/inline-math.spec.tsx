import React from 'react'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import InlineMath from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('render with custom className', () => {
    const code = 'x^2 + y^2 = z^2'
    const className = 'custom-inline-math'
    const wrapper = render(
      <InlineMath className={ className } value={ code } />
    )
    expect(wrapper.hasClass(className)).toEqual(true)
  })

  it('value is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(<InlineMath value={ value } />)
      }).toThrow(/Failed prop type: The prop `value` is marked as required/i)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLSpanElement>()
    const wrapper = mount(
      <InlineMath ref={ ref } data-value="waw" value=""/>
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const wrapper = mount(
      <InlineMath
        value="x^2 + y^2 = z^2"
        style={ { color: 'orange', fontSize: '16px' } }
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      yozora: {
        inlineMath: {
          padding: '2px',
          border: '1px solid blue',
          // margin: '0 2px',
          background: 'hsla(210deg, 13%, 12%, 0.05)',
          color: '#d81848',
        }
      }
    }

    const wrapper = mount(
      <ThemeProvider theme={ theme }>
        <InlineMath
          value="x^2 + y^2 = z^2"
          style={ { color: 'orange', fontSize: '16px' } }
        />
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
