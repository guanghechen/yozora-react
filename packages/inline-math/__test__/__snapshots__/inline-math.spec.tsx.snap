// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
<InlineMath
  style={
    Object {
      "color": "orange",
      "fontSize": "16px",
    }
  }
  value="x^2 + y^2 = z^2"
>
  <styled.span
    style={
      Object {
        "color": "orange",
        "fontSize": "16px",
      }
    }
  >
    <span
      className=""
      style={
        Object {
          "color": "orange",
          "fontSize": "16px",
        }
      }
    >
      <MathJaxNode
        formula="x^2 + y^2 = z^2"
        inline={true}
      />
    </span>
  </styled.span>
</InlineMath>
`;
